CatalogService.BLL/Exceptions/NotFoundException.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace CatalogService.BLL.Exceptions
+{
+    public class NotFoundException : Exception
```

Comment:
NotFoundException does not resolve level 2 maturity. It still returns 400 to a client. If a user tries to create an existing resource, we should return the 409 code. You must catch the exception in the controller or write asp.net core middle/filter.

Reply:
On that scenario client cannot create same resource because I changed the Id as Guid and I will create a new one every time.

Reply:
Not found is for Update, Delete and Get operations not create ones.

Reply:
I know BLL, DAL is not necessary but I wanted to implement with 3 tier and use AutoMapper.

Reply:
Create and Update model does not include Id anymore. When I finish everything I will write to you to check again.

Reply:
I finished and send last commit.